GetStream/stream-chat-swift
# [CIS-1062] Support custom localization

# What this PR do:
- Adds support for overriding default english (and possibly other) localizations in Apps using StreamChat.  Also adds documentation on how to use custom localization.
# How to test it
- Run the Demo Application in Czech language and check if the static texts are translated to Czech language.
# Where you can start
-  `Appearance.default.localizationProvider`, `SwiftGen` config.

# How this works:

`Appearance.default.localizationProvider` is simple closure, which defines how what format of localized strings will be used. Once the localizationProvider is set, it is used inside internal enum `L10n`, which will use the format for getting localized strings.
